### ISSUE TITLE
[BUG] Fixtures re-run when Playwright timeout is exceeded

### ISSUE THREAD

**Issue by dwalleck** [microsoft/playwright#9040]
**Context:**
- Playwright Version: 1.14.1
- Operating System: Windows
- Node.js version: 14.17.6
- Browser: All

**Code Snippet**

A full reproduction can be found [here](https://github.com/dwalleck/playwright-timeout-fixture-reproduction)

Given a `timeout` value of 5000 in `playwright.config.ts`:

```javascript
const test = base.extend<TestFixtures>({
    longOperation: async({}, use) => {
        const sleep = (ms: number) => {
            return new Promise(resolve => {
              setTimeout(resolve, ms)
            })
          }
        console.log('Starting long operation');
        await sleep(20000);
        console.log('Ending long operation');
        use('done')
    }
});
```
yields this output:

```
❯  npx playwright test example.test.ts
Using config at C:\Users\dwall\repos\playwright-timeout-fixture-reproduction\playwright.config.ts

Running 1 test using 1 worker

Starting long operation
Starting long operation
Starting long operation
Starting long operation
```
**Describe the bug**

I'm not sure if this is a bug or behavior that is not well defined. Some of my team members ran into an issue where their tests seemed to be in an infinite loop. On further investigation, I realized that one of our custom fixtures could exceed the Playwright timeout. My expectation is that an error would be thrown and the test would be halted. If this retry is intentional, it would be helpful to add something to the verbose logging that let the user know that fixture had timed out and was being restarted.

**Comment by mxschmitt** [comment]
You need to await the `use` function. See here: https://playwright.dev/docs/test-fixtures#with-fixtures
Did you set a retry in your config? https://playwright.dev/docs/test-retries

**Comment by dwalleck** [comment]
> You need to await the `use` function. See here: https://playwright.dev/docs/test-fixtures#with-fixtures (but this does not explain the behaviour).
> Did you enable retries in your config? https://playwright.dev/docs/test-retries

I did not enable retries in the config. I copied over my config to the repro and only removed a few entries such as baseUrl that were not needed.

**Comment by mxschmitt** [comment]
I can reproduce in version 1.14 but it seems fixed in 1.15. Please verify that, thank you!

**Comment by dwalleck** [comment]
I can verify it works now. Thanks!
